File: python/src/agent_api/types/safety_identifiers.py
from __future__ import annotations

from typing import Literal, TypedDict


class SafetyIdentifierPartition(TypedDict, total=False):
    object: Literal["safety_identifier"]
    workspace_id: str
    safety_identifier: str
    owner_user_id: str
    status: str
    created_at: int
    updated_at: int


class ListSafetyIdentifierPartitionsResponse(TypedDict, total=False):
    object: Literal["list"]
    data: list[SafetyIdentifierPartition]

File: javascript/src/types/safety-identifiers.ts
export interface SafetyIdentifierPartition {
  object: "safety_identifier";
  workspace_id: string;
  safety_identifier: string;
  created_by_user_id: string;
  status: string;
  created_at: number;
  updated_at: number;
}

export interface ListSafetyIdentifierPartitionsParams {
  page_size?: number;
  page_token?: string;
}

export interface ListSafetyIdentifierPartitionsResponse {
  object: "list";
  data: SafetyIdentifierPartition[];
}
